### PR TITLE
Added 3.7 to HelioPy Testing

### PR DESCRIPTION
**Pull request by abotiamnot**
3.7 support hasn't been added yet, but we should let this PR stay for the time being.

### CODE REVIEW

**Comment by abotiamnot** [comment]
`Recent Python development branches require OpenSSL 1.0.2+. As this library is not available for Trusty, 3.7-dev, 3.8-dev, and nightly do not work (or use outdated archive).`

From the [TravisCI Documentation](https://docs.travis-ci.com/user/languages/python/)

**Comment by dstansby** [comment]
xref https://github.com/travis-ci/travis-ci/issues/9815

**Comment by abotiamnot** [comment]
@dstansby Python 3.7 is now working with Travis!

**Comment by dstansby** [comment]
Yay! The tests are failing though 😢 . Looks like quite a few are taking a long time, and some are failing.

File: requirements/test.txt [hunk 1]
```diff
@@ -3,4 +3,3 @@
 pytest
 pytest-cov
 pytest-pep8
-tables
```
Comment (dstansby) [inline requirements/test.txt]: Why this change?

Comment (abotiamnot) [reply requirements/test.txt]: It seems to be failing when we add tables. I'll make the changes again to show them!

**Comment by abotiamnot** [comment]
Could this be happening due to the fact that most of the Python packages are not yet Python 3.7 compatible? When I try to install Sunpy, AstroPy or Matplotlib, they seem to downgrade my Python installation from 3.7 to 3.6.6

**Comment by dstansby** [comment]
If you remove the `--quiet` option in the pip install bit it might help see what's going on. (https://travis-ci.org/heliopython/heliopy/jobs/401854557#L643)

**Comment by dstansby** [comment]
I've opened an issue here: https://github.com/PyTables/PyTables/issues/702

**Comment by dstansby** [comment]
Ah, it looks like hdf5 isn't being installed, try adding `hdf5` to the conda installation line that currently says `- conda create --yes -n testenv python=$TRAVIS_PYTHON_VERSION `

**Comment by abotiamnot** [comment]
@dstansby Can you take a look at the new TravisCI Log?

**Comment by dstansby** [comment]
I used `conda install hdf5`locally and it worked fine, so might be worth trying that instead of `h5py`

**Comment by abotiamnot** [comment]
Thanks!

**Comment by dstansby** [comment]
Is there a reason that we need `sudo: required` on the python 3.7 build?

**Comment by abotiamnot** [comment]
I added a sudo: required so that 3.7 runs in an independent environment, and not a shared one. I was just playing around with this! I think this is the reason we had to add hd5 explicitly unlike the Python 3.6 build.

**Comment by dstansby** [comment]
One way to solve this (that I think PlasmaPy are doing) is to install a specific version of python when the conda environment is created, since that is the version of python that the tests run. (and just have both travis system python versions as 3.6)

**Comment by dstansby** [comment]
Going to close this in favour of the other PRs for testing on 3.7.
